subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/TestDataGenerator.java

```diff
@@ -163,4 +173,37 @@ public String getDate() {
             return commits;
         }
     }
+
+    static class BackgroundColor {
+        // https://www.w3schools.com/colors/colors_picker.asp
+        private static final String RED = "#ff0000";
+        private static final String ORANGE = "#ff8000";
+        private static final String LIGHT_GREEN = "#80ff00";
+        private static final String GREEN = "#00ff00";
+        private Map xaxis;
+        private String color;
+
+        private BackgroundColor(List<Number> xy) {
+            double index = xy.get(0).doubleValue();
+            double differencePercentage = xy.get(1).doubleValue();
+            xaxis = ImmutableMap.of("from", index - 0.5, "to", index + 0.5);
+            if (differencePercentage > 2) {
```

Comment:
If we use confidence here, I would say that anything < 80% confidence is a white background. Then we could scale from light red to darker red (probably with some opacity) evenly between 80-100%. We would do the same with "negative" confidence, i.e. an improvement, with green.